p2p/src/peers.rs

```diff
@@ -424,6 +424,14 @@ impl Peers {
 			} else if !peer.is_connected() {
 				debug!(LOGGER, "clean_peers {:?}, not connected", peer.info.addr);
 				rm.push(peer.clone());
+			} else {
+				let (stuck, diff) = peer.is_stuck();
+				if stuck && diff < self.adapter.total_difficulty() {
```

Comment:
Taking into account @jaspervdm 's comment: now, only kick out stuck peer when peer's difficulty is lower than own difficulty.